TM1py/Services/GitService.py

```diff
@@ -0,0 +1,179 @@
+
+    def git_status(self, **kwargs) -> Git:
+        """ Get GIT status, returns Git object
+        :param Username: GIT username
+        :param Password: GIT password
+        :param PublicKey: SSH public key, available from PAA V2.0.0.4
+        :param PrivateKey: SSH private key, available from PAA V2.0.0.4
+        :param Passphrase: Passphrase for decrypting private key, if set
+        """
+        url = "/api/v1/GitStatus"
+        body = json.dumps(kwargs)
+        response = self._rest.POST(url=url, data=body)
+
+        return self.get_git_from_response(response)
+
+    def git_push(self, execute: bool = False, **kwargs) -> Response:
+        """ Creates a gitpush plan, returns response
+        :param execute: Executes the plan right away if True
+        :param Branch: The branch which last commit will be used as parent commit for new branch. Must be empty if GIT repo is empty
+        :param NewBranch: If specified, creates a new branch and pushes the commit onto it. If not specified, pushes to the branch specified in "Branch"
+        :param Force: A flag passed in for evaluating preconditions
+        :param Message: Commit message
+        :param Author: Name of commit author
+        :param Email: Email of commit author
+        :param Username: GIT username
+        :param Password: GIT password
+        :param PublicKey: SSH public key, available from PAA V2.0.0.4
+        :param PrivateKey: SSH private key, available from PAA V2.0.0.4
+        :param Passphrase: Passphrase for decrypting private key, if set
+        """
+        url = "/api/v1/GitPush"
+        body = json.dumps(kwargs)
+        response = self._rest.POST(url=url, data=body)
+        if execute:
+            planid = json.loads(response.content).get('ID')
+            self.git_executeplan(planid=planid)
+
+        return response
+
+    def git_pull(self, execute: bool = False, **kwargs) -> Response:
+        """ Creates a gitpull plan, returns response
+        :param execute: Executes the plan right away if True
+        :param Branch: The name of source branch
+        :param Force: A flag passed in for evaluating preconditions
+        :param Username: GIT username
+        :param Password: GIT password
+        :param PublicKey: SSH public key, available from PAA V2.0.0.4
+        :param PrivateKey: SSH private key, available from PAA V2.0.0.4
+        :param Passphrase: Passphrase for decrypting private key, if set
+        """
+        url = "/api/v1/GitPull"
+        body = json.dumps(kwargs)
+        response = self._rest.POST(url=url, data=body)
+        if execute:
+            planid = json.loads(response.content).get('ID')
+            self.git_executeplan(planid=planid)
+
+        return response
+
+    def git_executeplan(self, planid) -> Response:
+        """ Executes a plan based on the planid
+        :param planid: GitPlan id
+        """
+
+        url = "/api/v1/GitPlans('{}')/tm1.Execute".format(planid)
```

Comment:
You can use the `format_url` function if there are characters in the `plan_id` that must be url-escaped